core-bundle/src/Picker/AbstractTablePickerProvider.php

```diff
@@ -226,36 +231,29 @@ protected function getPtableAndPid(string $table, string $value): array
         // Use the first value if array to find a database record
         $id = (int) explode(',', $value)[0];
 
-        if (!$value) {
-            return [null, null];
-        }
-
         $this->framework->initialize();
         $this->framework->createInstance(DcaLoader::class, [$table])->load();
 
-        $pid = null;
         $ptable = $GLOBALS['TL_DCA'][$table]['config']['ptable'] ?? null;
         $dynamicPtable = $GLOBALS['TL_DCA'][$table]['config']['dynamicPtable'] ?? false;
 
         if (!$ptable && !$dynamicPtable) {
             return [null, null];
         }
 
-        $qb = $this->connection->createQueryBuilder();
-        $qb->select('pid')->from($table)->where($qb->expr()->eq('id', $id));
+        $data = false;
 
-        if ($dynamicPtable) {
-            $qb->addSelect('ptable');
-        }
+        if ($id) {
+            $qb = $this->connection->createQueryBuilder();
+            $qb->select('pid')->from($table)->where($qb->expr()->eq('id', $id));
 
-        $data = $qb->execute()->fetch();
+            if ($dynamicPtable) {
+                $qb->addSelect('ptable');
+            }
 
-        if (false === $data) {
-            return [null, null];
+            $data = $qb->execute()->fetch();
         }
 
-        $pid = (int) $data['pid'];
-
         if ($dynamicPtable) {
             $ptable = $data['ptable'] ?: $ptable;
```

Comment:
IIRC, `$data` can be `false` at this point. Should we replace `?:` with `??`?

Reply:
In Contao 4.9, `ptable` can be an empty string: https://github.com/contao/contao/blob/2bc452ebc042a476285243a3c2371c1fa0203a9b/core-bundle/src/Resources/contao/dca/tl_content.php#L18

Reply:
```suggestion
            if (!empty($data['ptable'])) {
                $ptable = $data['ptable'];
            }
```